### PR TITLE
Added softmax operator

### PR DESCRIPTION
**Pull request by cliffburdick**

### CODE REVIEW

**Comment by luitjens** [comment]
Can you commit the benchmark too?

File: include/matx/transforms/reduce.h [hunk 1]
```diff
@@ -1173,6 +1210,103 @@ void __MATX_INLINE__ mean(OutType dest, const InType &in, const int (&dims)[D],
...
+  }
+
+  auto tmp_sum = make_tensor<typename InType::scalar_type>(red_shape, MATX_ASYNC_DEVICE_MEMORY, stream);
+  sum(tmp_sum, exp(permute(in, perm)), stream);
+
+  // With the sum calculated, we have a tensor that's not compatible in sizes with the new one for dividing.
+  // We need to clone the summed tensor on the appropriate dims for the final divide.
+  std::array<index_t, InType::Rank()> clone_dims;
+  int axis_ptr = 0;
+  for (int r = 0; r < InType::Rank(); r++) {
```
Comment (luitjens) [inline include/matx/transforms/reduce.h:1294]: unroll?

**Review by luitjens — APPROVED** [review]
should should probably also upload the benchmark you made.

**Comment by cliffburdick** [comment]
> Can you commit the benchmark too?

it's in the second commit

**Pull request merged**